Grading at 10/26/2024:21:59:39.269837
For commit ID 520b9f3ef74ba3c84fe25ae8156d105c30fcb1a8
Grading at Sat Oct 26 05:59:23 PM EDT 2024
Doing some basic checking for your submission

Step 1
Checking that story-step1 compiles
rm -f *~ story-step1 story-step2 story-step3  story-step1.o story-step2.o story-step3.o rand_story.o
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c rand_story.c
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c story-step1.c
gcc -o story-step1 story-step1.o rand_story.o provided.o
rm story-step1.o
The code compiled (good)
Executable story-step1 exists (good)
Checking with provided test case story.txt
Your file matched the expected output
Now running YOUR tests for step 1

1.1: Current test is story-step1 ./story.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

1.2: Current test is story-step1 ./story2.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

1.3: Current test is story-step1 ./test-step1/separate-line.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

1.4: Current test is story-step1 ./test-step1/nonexist.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

1.5: Current test is story-step1 ./test-step1/empty.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

1.6: Current test is story-step1 ./test-step1/newline.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

Step 2
Checking that story-step2 compiles
rm -f *~ story-step1 story-step2 story-step3  story-step1.o story-step2.o story-step3.o rand_story.o
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c rand_story.c
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c story-step2.c
gcc -o story-step2 story-step2.o rand_story.o provided.o
rm story-step2.o
The code compiled (good)
Executable story-step2 exists (good)
Now running YOUR tests for step 2

2.1: Current test is story-step2 ./words.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.2: Current test is story-step2 ./test-step2/double-colon.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.3: Current test is story-step2 ./test-step2/no-colon.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

2.4: Current test is story-step2 ./test-step1/nonexist.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

2.5: Current test is story-step2 ./test-step1/empty.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

2.6: Current test is story-step2 ./test-step1/newline.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

Step 3
Checking that story-step3 compiles
rm -f *~ story-step1 story-step2 story-step3  story-step1.o story-step2.o story-step3.o rand_story.o
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c rand_story.c
gcc -std=gnu99 -pedantic -Wall -Werror -ggdb3 -c story-step3.c
gcc -o story-step3 story-step3.o rand_story.o provided.o
rm story-step3.o
The code compiled (good)
Executable story-step3 exists (good)
Now running YOUR tests for step 3

3.1: Current test is story-step3 ./words.txt ./story.txt 
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

							      >	Once upon a time, there was a walrus who lived in a very pecu
							      >	cave. 
Your file did not match the expected output


3.2: Current test is story-step3 ./words.txt ./story2.txt
-----------------------------
It looks like you got different output from us on that case.
Here is the diff (yours on left, ours on right):

							      >	Once upon a time, there was a walrus. This walrus lived in a 
							      >	peculiar cave. One day, it left its peculiar cave and met a w
Your file did not match the expected output


3.3: Current test is story-step3 ./test-step1/nonexist.txt ./story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.4: Current test is story-step3 ./test-step1/empty.txt ./story.txt
-----------------------------
We considered this a failure case.
But you did not (exit status was EXIT_SUCCESS).

3.5: Current test is story-step3 ./test-step1/newline.txt ./story.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.6: Current test is story-step3 ./words.txt ./test-step1/nonexist.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.7: Current test is story-step3 ./test-step1/nonexist.txt ./test-step1/nonexist.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.8: Current test is story-step3 ./test-step1/empty.txt ./test-step1/nonexist.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.9: Current test is story-step3 ./test-step1/newline.txt ./test-step1/nonexist.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.10: Current test is story-step3 ./words.txt ./test-step1/empty.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind reported errors or memory leaks:
==3722336== Memcheck, a memory error detector
==3722336== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3722336== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3722336== Command: ./story-step3 ./words.txt ./test-step1/empty.txt
==3722336== Parent PID: 3722335
==3722336== 
==3722336== 
==3722336== HEAP SUMMARY:
==3722336==     in use at exit: 1,279 bytes in 21 blocks
==3722336==   total heap usage: 37 allocs, 16 frees, 10,919 bytes allocated
==3722336== 
==3722336== 503 (16 direct, 487 indirect) bytes in 1 blocks are definitely lost in loss record 6 of 8
==3722336==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3722336==    by 0x109CCC: buildCategory (rand_story.c:221)
==3722336==    by 0x1093FB: main (story-step3.c:11)
==3722336== 
==3722336== 776 (56 direct, 720 indirect) bytes in 1 blocks are definitely lost in loss record 8 of 8
==3722336==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3722336==    by 0x10951C: readAFile (rand_story.c:30)
==3722336==    by 0x1093EB: main (story-step3.c:10)
==3722336== 
==3722336== LEAK SUMMARY:
==3722336==    definitely lost: 72 bytes in 2 blocks
==3722336==    indirectly lost: 1,207 bytes in 19 blocks
==3722336==      possibly lost: 0 bytes in 0 blocks
==3722336==    still reachable: 0 bytes in 0 blocks
==3722336==         suppressed: 0 bytes in 0 blocks
==3722336== 
==3722336== For lists of detected and suppressed errors, rerun with: -s
==3722336== ERROR SUMMARY: 2 errors from 2 contexts (suppressed: 0 from 0)

3.11: Current test is story-step3 ./test-step1/nonexist.txt ./test-step1/empty.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.12: Current test is story-step3 ./test-step1/empty.txt ./test-step1/empty.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind was clean

3.13: Current test is story-step3 ./test-step1/newline.txt ./test-step1/empty.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.14: Current test is story-step3 ./words.txt ./test-step1/newline.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind reported errors or memory leaks:
==3722512== Memcheck, a memory error detector
==3722512== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3722512== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3722512== Command: ./story-step3 ./words.txt ./test-step1/newline.txt
==3722512== Parent PID: 3722511
==3722512== 
==3722512== 
==3722512== HEAP SUMMARY:
==3722512==     in use at exit: 1,543 bytes in 24 blocks
==3722512==   total heap usage: 41 allocs, 17 frees, 11,199 bytes allocated
==3722512== 
==3722512== 264 (24 direct, 240 indirect) bytes in 1 blocks are definitely lost in loss record 6 of 10
==3722512==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3722512==    by 0x10951C: readAFile (rand_story.c:30)
==3722512==    by 0x1093D4: main (story-step3.c:8)
==3722512== 
==3722512== 503 (16 direct, 487 indirect) bytes in 1 blocks are definitely lost in loss record 8 of 10
==3722512==    at 0x4848899: malloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3722512==    by 0x109CCC: buildCategory (rand_story.c:221)
==3722512==    by 0x1093FB: main (story-step3.c:11)
==3722512== 
==3722512== 776 (56 direct, 720 indirect) bytes in 1 blocks are definitely lost in loss record 10 of 10
==3722512==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3722512==    by 0x10951C: readAFile (rand_story.c:30)
==3722512==    by 0x1093EB: main (story-step3.c:10)
==3722512== 
==3722512== LEAK SUMMARY:
==3722512==    definitely lost: 96 bytes in 3 blocks
==3722512==    indirectly lost: 1,447 bytes in 21 blocks
==3722512==      possibly lost: 0 bytes in 0 blocks
==3722512==    still reachable: 0 bytes in 0 blocks
==3722512==         suppressed: 0 bytes in 0 blocks
==3722512== 
==3722512== For lists of detected and suppressed errors, rerun with: -s
==3722512== ERROR SUMMARY: 3 errors from 3 contexts (suppressed: 0 from 0)

3.15: Current test is story-step3 ./test-step1/nonexist.txt ./test-step1/newline.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

3.16: Current test is story-step3 ./test-step1/empty.txt ./test-step1/newline.txt
-----------------------------
Your output looks good. Lets check valgrind
Valgrind reported errors or memory leaks:
==3722602== Memcheck, a memory error detector
==3722602== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==3722602== Using Valgrind-3.18.1 and LibVEX; rerun with -h for copyright info
==3722602== Command: ./story-step3 ./test-step1/empty.txt ./test-step1/newline.txt
==3722602== Parent PID: 3722601
==3722602== 
==3722602== 
==3722602== HEAP SUMMARY:
==3722602==     in use at exit: 264 bytes in 3 blocks
==3722602==   total heap usage: 10 allocs, 7 frees, 9,656 bytes allocated
==3722602== 
==3722602== 264 (24 direct, 240 indirect) bytes in 1 blocks are definitely lost in loss record 2 of 2
==3722602==    at 0x484DCD3: realloc (in /usr/libexec/valgrind/vgpreload_memcheck-amd64-linux.so)
==3722602==    by 0x10951C: readAFile (rand_story.c:30)
==3722602==    by 0x1093D4: main (story-step3.c:8)
==3722602== 
==3722602== LEAK SUMMARY:
==3722602==    definitely lost: 24 bytes in 1 blocks
==3722602==    indirectly lost: 240 bytes in 2 blocks
==3722602==      possibly lost: 0 bytes in 0 blocks
==3722602==    still reachable: 0 bytes in 0 blocks
==3722602==         suppressed: 0 bytes in 0 blocks
==3722602== 
==3722602== For lists of detected and suppressed errors, rerun with: -s
==3722602== ERROR SUMMARY: 1 errors from 1 contexts (suppressed: 0 from 0)

3.17: Current test is story-step3 ./test-step1/newline.txt ./test-step1/newline.txt
-----------------------------
Great: we both considered this an error
Lets check if it valgrinds:
Valgrind was clean

Step 4
Checking that story-step4 compiles
rm -f *~ story-step1 story-step2 story-step3  story-step1.o story-step2.o story-step3.o rand_story.o
The code did not compile!

Overall Grade: NO GRADE YET
